lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
-
-                if (item["last_price_date"] && item["last_price_date"].length === 10) {
-                    item["last_price_date"] = new Date(item["last_price_date"]);
-                }
-            }
+    const results: any[] = [];
+    const fields = ['order_timestamp', 'exchange_timestamp', 'created', 'last_instalment', 'fill_timestamp'];
 
-            return parsedData;
+    for (const item of list) {
+        for (const field of fields) {
+            if (item[field] && item[field].length === 19) {
+                item[field] = new Date(item[field]);
+            }
         }
 
-        return data;
+        results.push(item);
     }
 
-    function _get(route, params?, responseType?, responseTransformer?, isJSON = false) {
-        return request(route, "GET", params || {}, responseType, responseTransformer, isJSON);
+    if (data.data instanceof Array) {
+        data.data = results;
+    } else {
+        data.data = results[0];
     }
 
-    function _post(route, params, responseType?, responseTransformer?, isJSON = false, queryParams: any =null) {
-        return request(route, "POST", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+    return data;
+}
 
-    function _put(route, params, responseType?, responseTransformer?, isJSON = false, queryParams=null) {
-        return request(route, "PUT", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+function parseHistorical(jsonData: AnyObject): AnyObject {
+    // Return if its an error
+    if (jsonData.error_type) return jsonData;
+
+    const results = [] as any[];
+    for (let i = 0; i < jsonData.data.candles.length; i++) {
+        const d = jsonData.data.candles[i];
+        const c: AnyObject = {
+            'date': new Date(d[0]),
+            'open': d[1],
+            'high': d[2],
+            'low': d[3],
+            'close': d[4],
+            'volume': d[5]
+        }
+
+        // Add OI field if its returned
+        if (d[6]) {
+            c['oi'] = d[6]
+        }
 
-    function _delete(route, params, responseType?, responseTransformer?, isJSON = false) {
-        return request(route, "DELETE", params || {}, responseType, responseTransformer, isJSON);
+        results.push(c);
     }
 
-    function request(route, method, params, responseType?, responseTransformer?, isJSON?, queryParams?) {
-        // Check access token
-        if (!responseType) responseType = "json";
-        let uri = routes[route];
+    return { 'data': results };
+}
 
-        // Replace variables in "RESTful" URLs with corresponding params
-        if(uri.indexOf("{") !== -1) {
-            let k;
-            for(k in params) {
-                if(params.hasOwnProperty(k)) {
-                    uri = uri.replace("{" + k + "}", params[k]);
-                }
+function transformInstrumentsResponse(data: any, headers: AnyObject): unknown[] | any {
```

Comment:
It is advisable to designate the `transformInstrumentsResponse` function as a private function.